### PR TITLE
Sample dapp screens

### PR DESCRIPTION
**Pull request by kacperoak**
Based on changes from PR: https://github.com/WalletConnect/WalletConnectKotlinV2/pull/832

- Pairing selection screen
- Add session screen and selected account in sample dapp
- handle dapp events

### CODE REVIEW

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/domain/PushDappDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    override fun onPushRejected(rejection: Push.Dapp.Event.Rejected) {
+    }
+
+    override fun onDelete(pushDelete: Push.Dapp.Event.Delete) {
+    }
+
+    override fun onError(error: Push.Model.Error) {
+    }
```
Comment (jakubuid) [inline samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/domain/PushDappDelegate.kt:36]: What about `onPushRejected` `onDelete` and `onError` events?

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/data/explorer/ExplorerListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.JsonQualifier
+import com.squareup.moshi.ToJson
+import com.walletconnect.sample.dapp.web3modal.network.model.ExplorerWalletResponse
+
+@Retention(AnnotationRetention.RUNTIME)
+@JsonQualifier
+annotation class MapToList
+
+class ExplorerMapToListAdapter {
```
Comment (jakubuid) [inline samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/data/explorer/ExplorerListAdapter.kt:12]: Just to be sure that we're on the same page. This class and all components related to explorer will be in the web3.modal module within next PRs, right? Also, the same applies to UI web3.modal components

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/ui/Web3ModalEvents.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.walletconnect.sample.dapp.web3modal.ui
+
+sealed class Web3ModalEvents {
+    object SessionApproved: Web3ModalEvents()
+    object SessionRejected: Web3ModalEvents()
+    object NoAction: Web3ModalEvents()
```
Comment (jakubuid) [inline samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/ui/Web3ModalEvents.kt:6]: Looks like we can add in the future more but not for now

**Review by jakubuid — APPROVED** [review]

**Review by TalhaAli00 — APPROVED** [review]

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/domain/PushDappDelegate.kt (hunk 1 above)
Comment (kacperoak) [reply samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/domain/PushDappDelegate.kt:36]: this has been moved from :sign:app. So this is a hard question for me :D

File: samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/ui/Web3ModalEvents.kt (hunk 3 above)
Comment (kacperoak) [reply samples/dapp/src/main/kotlin/com/walletconnect/sample/dapp/web3modal/ui/Web3ModalEvents.kt:6]: Yes, that's my plan

**Pull request merged**